### PR TITLE
Factor out TestExponentialGrowthPartitionMemoryEstimator.makeFactory

### PR DESCRIPTION
**Pull request by dekimir**
## Description

DRY some copy-pasted code in TestExponentialGrowthPartitionMemoryEstimator.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: core/trino-main/src/test/java/io/trino/execution/scheduler/faulttolerant/TestExponentialGrowthPartitionMemoryEstimator.java [hunk 1]
```diff
@@ -55,14 +56,20 @@ public class TestExponentialGrowthPartitionMemoryEstimator
         throw new RuntimeException("should not be used");
     };
 
-    @Test
-    public void testDefaultInitialEstimation()
+    private ExponentialGrowthPartitionMemoryEstimator.Factory estimatorFactory;
```
Comment (martint) [inline core/trino-main/src/test/java/io/trino/execution/scheduler/faulttolerant/TestExponentialGrowthPartitionMemoryEstimator.java:59]: This is an anti-pattern. It makes the test class stateful and requires all tests to run in sequence. If you want to reuse the code to create the object, extract it into a method that creates a new instance it and returns it. Then call it from each of the individual test methods:

```java
private static ExponentialGrowthPartitionMemoryEstimator.Factory createEstimator()
{
    return new ExponentialGrowthPartitionMemoryEstimator.Factory(...);
}

@Test
public void testDefaultInitialEstimation()
{
    ExponentialGrowthPartitionMemoryEstimator.Factory estimatorFactory = createEstimator();
    ...
}
```

**Review by martint — CHANGES_REQUESTED** [review]

File: core/trino-main/src/test/java/io/trino/execution/scheduler/faulttolerant/TestExponentialGrowthPartitionMemoryEstimator.java (hunk 1 above)
Comment (dekimir) [reply core/trino-main/src/test/java/io/trino/execution/scheduler/faulttolerant/TestExponentialGrowthPartitionMemoryEstimator.java:59]: Can you help me understand what's different here from every other `@BeforeEach` in the codebase?

Comment (martint) [reply core/trino-main/src/test/java/io/trino/execution/scheduler/faulttolerant/TestExponentialGrowthPartitionMemoryEstimator.java:59]: It's not, but that's a pattern we want to move away from wherever possible, and certainly avoid introducing new instances unless absolutely necessary.

Comment (dekimir) [reply core/trino-main/src/test/java/io/trino/execution/scheduler/faulttolerant/TestExponentialGrowthPartitionMemoryEstimator.java]: Rewrote as suggested.

**Pull request merged**